ts/components/messages/MessageCTABar.tsx

```diff
@@ -231,24 +265,102 @@ class MessageCTABar extends React.PureComponent<Props, State> {
       .catch(_ =>
         showToast(I18n.t("messages.cta.reminderAddFailure"), "danger")
       );
+
+  private confirmSaveCalendarEventAlert = (
+    calendar: Calendar,
+    message: CreatedMessageWithContent,
+    dueDate: Date,
+    title: string,
+    eventId: string
+  ) =>
+    Alert.alert(
+      I18n.t("messages.cta.reminderAlertTitle"),
+      I18n.t("messages.cta.reminderAlertDescription"),
+      [
+        {
+          text: I18n.t("global.buttons.cancel"),
+          style: "cancel"
+        },
+        {
+          text: I18n.t("messages.cta.reminderAlertKeep"),
+          style: "default",
+          onPress: () =>
+            this.setState(
+              {
+                isEventInDeviceCalendar: true
+              },
+              () => {
+                // Add the calendar event to the store
+                this.props.addCalendarEvent({
+                  messageId: message.id,
+                  eventId
+                });
+              }
+            )
+        },
+        {
+          text: I18n.t("messages.cta.reminderAlertAdd"),
+          style: "default",
+          onPress: () =>
+            this.saveCalendarEvent(calendar, message, dueDate, title)
+        }
+      ],
+      { cancelable: false }
+    );
+
+  private addCalendarEventToDeviceCalendar = (
+    message: CreatedMessageWithContent,
+    dueDate: Date
+  ) => (calendar: Calendar) => {
+    const title = I18n.t("messages.cta.reminderTitle", {
+      title: message.content.subject
+    });
+    // tslint:disable-next-line: no-commented-code
+    const { preferredCalendar } = this.props;
+
+    this.props.hideModal();
+
+    if (preferredCalendar === undefined) {
+      this.props.preferredCalendarSaveSuccess(calendar);
+    }
+
+    // tslint:disable-next-line: no-floating-promises
+    this.searchEventInCalendar(dueDate, title)
```

Comment:
There is a floating promise in it (**this.saveCalendarEvent** returns a Promise)
You could rewrite this function as follows

```typescript
this.searchEventInCalendar(dueDate, title)
      .then(mayBeEventId =>
        mayBeEventId.foldL(
          async () => {
            await this.saveCalendarEvent(calendar, message, dueDate, title);
          },
          async eventId => {
            this.confirmSaveCalendarEventAlert(
              calendar,
              message,
              dueDate,
              title,
              eventId
            );
          }
        )
      )
      .catch(() => this.saveCalendarEvent(calendar, message, dueDate, title));
```